code/parse/sexp_container.cpp

```diff
@@ -787,3 +787,261 @@ const char *sexp_container_CTEXT(int node)
 	// we've hit sanity limit; give up
 	return Empty_str;
 }
+
+// status SEXPs
+int sexp_is_container_empty(int node)
+{
+	const char *container_name = CTEXT(node);
+	const auto *p_container = get_sexp_container(container_name);
+
+	if (!p_container) {
+		Warning(LOCATION, "Is-container-empty called on nonexistent container %s.", container_name);
+		return SEXP_FALSE;
```

Comment:
I kept going back and forth about whether this invalid data should return true or false here.  My use case for this sexp would be to check if a container which contains a list or queue of items that I need to do *something* to has contents before I try to do things with the items in it.  So I would probably constantly use `!(sexp_is_container_empty)`.  But others could use it to try to tell if something needs to be added to that container, because it's some container that the mission needs to be doing something with constantly.

There doesn't seem to be a good way to "foolproof" this for users during mission testing. I can't think of any way to gracefully handle it not existing since we can only return true or false, and either can be the expected or desired result of a container not existing.

I think there are two ways to improve this, if desired:

A) Change to an error, or
B) Add that ignoring the warning will lead to undesired behavior.

In the future, maybe it would be good to add a SEXP_ERROR, that prevents SEXPs from evaluating in situations like this and then logs the error in the debug log.

Reply:
Good thoughts!

Arguments that should be a variable name (`OPF_VARIABLE_NAME`) are handled strictly. The value *must* be a name of a known variable: `check_sexp_syntax()` rejects anything else.

However, @Karajorma designed container SEXPs to be much more flexible. An argument that should be a container name (`OPF*CONTAINER_NAME`) *doesn't* have to be a string literal. It can be a special argument or a usage of Replace Variable or even Replace Container Data.

That's why we have to do this runtime check on container name arguments in the first place.

Although you could make a case for `Error()` in this situation, Kara chose to use `Warning()` when he wrote the original code.

I'll double-check if that was intended behavior, and if so, what the reasoning was.

Reply:
As for return value here, my vote is still `SEXP_FALSE`. A nonexistent container can't be empty.

We could argue it's vacuously true and thus return `SEXP_TRUE`, but that's definitely not intuitive behavior for non-mathy people,. 😜

Reply:
Yup, and the hope is that the mission tester will heed the warning and fix the mission. So it should be fine either way.

Reply:
Per discussion with Kara, we've decided to leave the messages as warnings but print them to the mission log as well. I've pushed a commit.